tests/wallet/nft_wallet/test_nft_wallet.py

```diff
@@ -320,17 +332,28 @@ async def test_nft_wallet_creation_and_transfer(self_hostname: str, two_wallet_n
     )
     assert compute_memos(txs[0].spend_bundle)
 
+    await ensure_wallet_sync()
+
+    # We wish for the wallet to be in a specific state at this point after having
+    # become synced above.
     for i in range(1, num_blocks):
         await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph1))
 
-    await time_out_assert(30, wallet_node_0.wallet_state_manager.lock.locked, False)
```

Comment:
was this a misguided attempt at, essentially, waiting for the wallet to be synced?

Reply:
my read is this: this lock is used when the wallet is in long sync mode, so i think this was trying to ensure that we don't go into the next phase if a long sync might be in progress and waiting at an io transaction.  I'm certain there's no reason to do this if wait_for_wallet_synced is used.